metrics/web/test/auth/presentation/pages/loading_page_test.dart

```diff
@@ -81,12 +96,72 @@ void main() {
     );
 
     testWidgets(
-      "displays the PlatformBrightnessObserver widget",
-      (tester) async {
+      "does not redirect from the loading page if a user is logged in and the remote config is not initialized",
+      (WidgetTester tester) async {
+        final authNotifier = AuthNotifierMock();
+        final instantConfigNotifier = InstantConfigNotifierMock();
+        final navigatorObserver = _NavigatorObserverMock();
+
+        await tester.pumpWidget(
+          _LoadingPageTestbed(
+            authNotifier: authNotifier,
+            instantConfigNotifier: instantConfigNotifier,
+            navigatorObserver: navigatorObserver,
+          ),
+        );
+
+        reset(navigatorObserver);
+
+        when(authNotifier.isLoggedIn).thenReturn(true);
+        when(authNotifier.isLoading).thenReturn(false);
+        when(instantConfigNotifier.isLoading).thenReturn(true);
+
+        authNotifier.notifyListeners();
+
+        await mockNetworkImagesFor(() {
+          return tester.pump();
+        });
+
+        verifyNever(navigatorObserver.didPush(any, any));
+      },
+    );
+
+    testWidgets(
+      "does not redirect from the loading page if a user is not logged in and the remote config is not initialized",
+      (WidgetTester tester) async {
+        final authNotifier = AuthNotifierMock();
+        final instantConfigNotifier = InstantConfigNotifierMock();
+        final navigatorObserver = _NavigatorObserverMock();
+
         await tester.pumpWidget(
-          const _LoadingPageTestbed(),
+          _LoadingPageTestbed(
+            authNotifier: authNotifier,
+            instantConfigNotifier: instantConfigNotifier,
+            navigatorObserver: navigatorObserver,
+          ),
         );
 
+        reset(navigatorObserver);
+
+        when(authNotifier.isLoggedIn).thenReturn(false);
+        when(authNotifier.isLoading).thenReturn(false);
+        when(instantConfigNotifier.isLoading).thenReturn(true);
+
+        authNotifier.notifyListeners();
+
+        await mockNetworkImagesFor(() {
+          return tester.pump();
+        });
+
+        verifyNever(navigatorObserver.didPush(any, any));
```

Comment:
I suppose we can do it without any observers just verifying that `find.byType(LoadingPage)` finds one widget. And if the page pushes a new one then we can check that `find.byType(LoadingPage)` finds nothing.